src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -20,6 +22,24 @@ export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWiza
             if (!result) {
                 context.version = await promptForFuncVersion(context);
             } else {
+                const endofLifeDate = result.minorVersion.stackSettings.linuxRuntimeSettings?.endOfLifeDate;
+                const sixMonthsFromNow = new Date();
+                sixMonthsFromNow.setMonth(sixMonthsFromNow.getMonth() + 6);
+                const settingKey: string = 'endOfLifeWarning';
+                if (getWorkspaceSetting<boolean>('endOfLifeWarning')) {
+                    if (endofLifeDate) {
+                        if (compareDates(endofLifeDate, sixMonthsFromNow)) {
+                            const message = localize('endOfLife', "The chosen runtime stack has an end of support deadline coming up. After the deadline, function apps can be created and deployed, and existing apps continue to run. However, your apps won't be eligible for new features, security patches, performance optimizations, and support until you upgrade them");
```

Comment:
nit: line is really long 😭

Reply:
Yeah I was also thinking that. I was using the verbiage that they said to use in the original issue, but it does make it quite long.

Reply:
Yeah we should just split the string up into multiple lines